compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2574,12 +2574,18 @@ object Parsers {
       })
     }
 
-    /** TypeCaseClause     ::= ‘case’ InfixType ‘=>’ Type [nl]
+    /** TypeCaseClause     ::= ‘case’ (InfixType | ‘_’) ‘=>’ Type [nl]
      */
     def typeCaseClause(): CaseDef = atSpan(in.offset) {
       val pat = inSepRegion(InCase) {
         accept(CASE)
-        infixType()
+        in.token match {
+          case USCORE if in.lookahead.isArrow =>
+            in.skipToken()
+            scalaAny
```

Comment:
We usually avoid generating code that does not exist like that in source in Parser, since it will confuse IDEs. Why not `Ident(nme.WILDCARD)`?